package/cluster/eks/composition.yaml

```diff
@@ -251,3 +273,27 @@ spec:
                 fmt: "%s-ekscluster"
       readinessChecks:
         - type: None
+    - base:
+        apiVersion: apiextensions.crossplane.io/v1alpha1
+        kind: EnvironmentConfig
+      name: irsa-environmentconfig
+      patches:
+      - fromFieldPath: spec.id
+        toFieldPath: metadata.labels[clusterRef]
+      - fromFieldPath: status.eks.oidcUri
+        toFieldPath: data.oidcHost
+      - fromFieldPath: status.eks.accountId
+        toFieldPath: data.accountId
+      - type: CombineFromComposite
+        toFieldPath: data.oidcArn
```

Comment:
It's non-blocking to this PR, but I will note it here

With `--enable-composition-webhook-schema-validation` active, the validation webhook fails with

```
compositeresourcedefinition.apiextensions.crossplane.io/xeks.aws.platformref.upbound.io unchanged
The Composition "xeks.aws.platformref.upbound.io" is invalid:
* spec.resources[13].patches[1].toFieldPath: Invalid value: "data.oidcHost": field path "data.oidcHost" has an unsupported type ""
* spec.resources[13].patches[2].toFieldPath: Invalid value: "data.accountId": field path "data.accountId" has an unsupported type ""
* spec.resources[13].patches[3].toFieldPath: Invalid value: "data.oidcArn": field path "data.oidcArn" has an unsupported type ""
```

CC @phisco

Reply:
Sorry, just saw this, I’ll dig into this tomorrow, would still appreciate a dedicated issue with a few steps to reproduce it either using this PR or a simplified example 🙏

Reply:
add a dedicated issue: https://github.com/crossplane/crossplane/issues/4344

Reply:
@phisco  thanks for https://github.com/crossplane/crossplane/pull/4346

Reply:
thanks to you both for providing such valuable feedback! 🙏